apm-lambda-extension/Makefile

```diff
@@ -5,6 +5,10 @@ DOCKER_IMAGE_NAME = observability/apm-lambda-extension
 DOCKER_REGISTRY = docker.elastic.co
 AGENT_VERSION = $(shell echo $${BRANCH_NAME} | cut -f 2 -d 'v')
 
+# Add support for SOURCE_DATE_EPOCH and reproducble buils
+# See https://reproducible-builds.org/specs/source-date-epoch/
+SOURCE_DATE_EPOCH ?= 0
```

Comment:
Should we default to using the latest commit timestamp (`$(git log -1 --pretty=%ct)`)?

Reply:
I think this makes sense and the spec seems to allow it:

> The value MUST be reproducible (deterministic) across different executions of the build, depending only on the source code. It SHOULD be set to the last modification time of the source, incorporating any packaging-specific modifications.